api/pkg/apis/v1alpha1/providers/target/mqtt/mqtt_test.go

```diff
@@ -280,13 +310,73 @@ func TestApply(t *testing.T) {
 		}
 	}
 
-	_, err = provider.Apply(context.Background(), model.DeploymentSpec{
+	deploymentSpec := model.DeploymentSpec{
+		SolutionName: "test-solution",
+		Solution:     model.SolutionState{},
 		Instance: model.InstanceState{
-			Spec: &model.InstanceSpec{},
+			Spec: &model.InstanceSpec{
+				Name:        "test-instance",
+				DisplayName: "test-instance",
+				Solution:    "test-solution",
+				Target: model.TargetSelector{
+					Name: "test-target",
+				},
+			},
+		},
+		Targets: map[string]model.TargetState{
+			"test-target": {
+				Spec: &model.TargetSpec{
+					DisplayName: "test-target",
+					Components: []model.ComponentSpec{
+						{
+							Name: "test-component",
+							Type: "test-component",
+						},
+					},
+					Topologies: []model.TopologySpec{
+						{
+							Bindings: []model.BindingSpec{
+								{
+									Role:     "test-target",
+									Provider: "providers.target.mqtt",
+									Config: map[string]string{
+										"name":          MQTTName,
+										"brokerAddress": MQTTBrokerAddress,
+										"clientID":      MQTTClientID,
+										"requestTopic":  MQTTRequestTopic,
+										"responseTopic": MQTTResponseTopic,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Assignments: map[string]string{
+			"test-component": "{test-target}",
 		},
-	}, model.DeploymentStep{}, false) //TODO: this is probably broken: the step should contain at least a component
+	}
+
+	stepSpec := model.DeploymentStep{
+		Target: "test-target",
+		Components: []model.ComponentStep{{
+			Action: "update",
+			Component: model.ComponentSpec{
+				Name: "test-component",
+				Type: "test-component",
+			},
+		}},
+	}
+
+	ret, err := provider.Apply(context.Background(), deploymentSpec, stepSpec, false) //TODO: this is probably broken: the step should contain at least a component
```

Comment:
Can we please remove the TODO comment? Then I think we are ready to merge.

Reply:
Good catch! I removed this TODO comment.